Tasks/Common/SonarQubeHelpers/SonarQubeHelper.ps1

```diff
@@ -268,11 +263,49 @@ function IsFeatureEnabled
     return $enabledByDefault        
 }
 
+#
+# Exit if the build is a PR build and PRCA is not enabled
+#
 function ExitOnPRBuild
 {    
     if ((IsPrBuild) -and !(IsFeatureEnabled "SQPullRequestBot" $false))
     {
         Write-Host "SonarQube analysis is disabled during builds triggered by pull requests. Set a build variable named 'SQPullRequestBot' to 'true' to have the task post code analysis issues as comments in the PR. More information at http://go.microsoft.com/fwlink/?LinkID=786316"
         exit
     } 
+}
+
+#
+# Compares the SonarQube server version with 5.2. Returns 0 is identical, >0 if greater, <0 if lower.
+# Note that 5.2 introduces the async analysis mode resulting in major API changes. 
+function CompareSonarQubeVersionWith52
+{
+    $versionString = GetOrFetchSonarQubeVersionString
+    
+    # Strip out '-SNAPSHOT' if it is present in version (developer versions of SonarQube might return version in this format: 5.2-SNAPSHOT)
+    $sqServerVersion = ([string]$versionString).split("-")[0]
+
+    $sqVersion = New-Object -TypeName System.Version -ArgumentList $sqServerVersion
+    $sqVersion5dot2 = New-Object -TypeName System.Version -ArgumentList "5.2"
+    
+    return $sqVersion.CompareTo($sqVersion5dot2)
+}
+
+#
+# Helper that returns the version number of the SonarQube server
+#
+function GetOrFetchSonarQubeVersionString
+{         
+    $versionString = GetTaskContextVariable "MSBuild.SonarQube.ServerVersion"
```

Comment:
So this is an internal working variable that isn't expected to be set on entry, and isn't expected to be relied on by subsequent external tasks? The variable is used an published in the same way as the other variables that we are making "public" / documenting for other to use. It would be better if the name reflected this (e.g. by including "internal").